specification/customer-insights/resource-manager/Microsoft.CustomerInsights/2017-04-26/customer-insights.json

```diff
@@ -3149,6 +3149,378 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/predictions/{predictionName}": {
+      "put": {
+        "tags": [
+          "Predictions"
+        ],
+        "operationId": "Predictions_CreateOrUpdate",
+        "x-ms-examples": {
+          "Predictions_CreateOrUpdate": {
+            "$ref": "./examples/PredictionsCreateOrUpdate.json"
+          }
+        },
+        "description": "Creates a Prediction or updates an existing Prediction in the hub.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "hubName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the hub."
+          },
+          {
+            "name": "predictionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Prediction.",
+            "pattern": "^[a-zA-Z][a-zA-Z0-9_]+$",
+            "minLength": 1,
+            "maxLength": 512
```

Comment:
If chances of pattern changes are there, I think it is better to remove the validation from client side. Otherwise you will need to update the swagger each time there is a change in 
the pattern in server side and regenerate a new version of SDK to accommodate this. What do you think?